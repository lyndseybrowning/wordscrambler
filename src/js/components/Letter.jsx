import React from 'react';
import PropTypes from 'prop-types';

const Letter = ({ letter }) => (
  <span className="c-letter">{letter}</span>
);

Letter.propTypes = {
  letter: PropTypes.string.isRequired,
};

export default Letter;
