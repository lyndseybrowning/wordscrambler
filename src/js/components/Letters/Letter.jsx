import React, { Component } from 'react';
import PropTypes from 'prop-types';
import config from '../../scripts/config';

class Letter extends Component {
  state = {
    selected: false,
  };

  toggleSelected = () => {
    this.setState({
      selected: !this.state.selected,
    });
  };

  render() {
    const letter = this.props.letter.toUpperCase();
    const cls = this.state.selected ? 'c-letter c-letter--selected' : 'c-letter';

    return (
      <div className={`o-grid__item u-1/${this.props.width}`}>
        <button
          className={cls}
          data-test-id="letter"
          onClick={this.toggleSelected}
          onTouchStart={this.toggleSelected}
        >
          {letter}
        </button>
      </div>
    );
  }
}

Letter.defaultProps = {
  width: config.NUM_LETTERS,
};

Letter.propTypes = {
  letter: PropTypes.string.isRequired,
  width: PropTypes.number,
};

export default Letter;
