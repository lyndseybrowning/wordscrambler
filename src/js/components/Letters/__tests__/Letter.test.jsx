import React from 'react';
import Letter from '../Letter';

it('should render correctly', () => {
  const letter = shallow(<Letter letter="A" />);

  expect(letter).toMatchSnapshot();
});

it('should render the provided letter', () => {
  const letter = shallow(<Letter letter="B" />);

  expect(letter.find('[data-test-id="letter"]').text()).toBe('B');
});

it('should render the letter to uppercase', () => {
  const letter = shallow(<Letter letter="b" />);

  expect(letter.find('[data-test-id="letter"]').text()).toBe('B');
});

it('should toggle the modifier class when clicked or touched', () => {
  const letter = shallow(<Letter letter="L" />);
  const button = letter.find('[data-test-id="letter"]');

  expect(button.hasClass('c-letter--selected')).toBe(false);

  button.simulate('click');

  expect(letter.find('[data-test-id="letter"]').hasClass('c-letter--selected')).toBe(true);

  button.simulate('touchStart');

  expect(letter.find('[data-test-id="letter"]').hasClass('c-letter--selected')).toBe(false);
});
