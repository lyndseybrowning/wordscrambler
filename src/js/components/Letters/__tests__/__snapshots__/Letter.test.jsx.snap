// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="o-grid__item u-1/6"
>
  <button
    className="c-letter"
    data-test-id="letter"
    onClick={[Function]}
    onTouchStart={[Function]}
  >
    A
  </button>
</div>
`;
